startup/94-multi_spectra.py

```diff
@@ -0,0 +1,773 @@
+from bluesky.simulators import summarize_plan
+
+
+def ERamp(dets, start, stop, velocity, streamname='primary'):
+    yield from count(dets)
+
+
+hw = hw()
+specs = hw.det2
+specs.name = 'specs'
+specs.gain = Signal(name='specs_gain')
+specs.decade = Signal(name='specs_decade')
+specs.low_energy = Signal(name='specs_low_energy')
+specs.high_energy = Signal(name='specs_high_energy')
+specs.step_size = Signal(name='specs_step_size')
+
+
+def _set_mode(val):
+    yield from mv(specs.acquisition_mode, val)
+
+
+specs.set_mode = _set_mode
+specs.acquisition_mode = Signal(name='specs_acquisition_mode')
+specs.num_acquisitions = Signal(name='specs_num_acquisitions')
+vortex = hw.det1
+vortex.name = 'vortex'
+vortex.gain = Signal(name='vortex_gain')
+vortex.decade = Signal(name='vortex_decade')
+
+
+class Pgm():
+    def __init__(self, name):
+        self.energy = Signal(name='pgm_energy')
+        self.name = name
+
+    def reset_fbl(self, *args, **kwargs):
+        yield from sleep(10)
+
+
+pgm = Pgm('pgm')
+motor = hw.motor
+motor1 = hw.motor1
+motor2 = hw.motor2
+motor3 = hw.motor3
+det = hw.det
+
+# above are the temporary objects for testing.
+
+
+def _str_to_obj(str_ref):
+    '''Converts a string ref to an ``ophyd.Device`` object reference.
+
+    Parameters
+    ----------
+    str_ref : str
+        A string reference to an ``ophyd.Device`` object, should have the form
+        'attribute', 'parent.attribute', 'grandparent.parent.attribute' or
+        'ancestor.(...).parent.attribute'.
+
+    Returns
+    -------
+    obj : object
+        The ``ophyd.Device`` object referenced by the string
+    '''
+    parent, _, attrs = str_ref.partition('.')
+    obj = ip.user_ns[parent]
+    if attrs:
+        obj = getattr(obj, attrs)
+    return obj
+
+
+def _move_from_dict(move_dict):
+    '''Returns a plan to moves a set of motors mapped to positions.
+
+    This expects a dictionary whose keys are strings containing an
+    ``ophyd.Device`` reference of the form ``parent.attribute`` or
+    ``grandparent.parent.attribute``, where the attribute is the
+    ``ophyd.Device`` to be moved. The values of the dictionary are the
+    positions to move to.
+
+    Parameters
+    ----------
+    move_dict, Dict[obj, value]
+        A dictionary mapping ``ophyd.Device``s to be moved to positions
+        to move to. See above for a description of this dictionary.
+    '''
+    settings_list = []
+    for key, value in move_dict.items():
+        obj = _str_to_obj(key)
+        settings_list.extend([obj, value])
+
+    return (yield from mv(*settings_list))
+
+
+def ios_multiscan_plan_factory_wrapper(scans):
+    '''Returns a plan that will perform multiple scans in order.
+
+    This generates a plan to set each value required to generate a scan at IOS
+    for each scan in 'scans', it is a wrapper around ios_multiscan_plan_factory
+    which parses the plan arguments string loaded from the excell file first.
```

Comment:
I saw multiple cases of this negligible typo. Fell free to ignore or apply it to other occurrences.
```suggestion
    which parses the plan arguments string loaded from the Excel file first.
```